lib/screen/detail/preview/keyboard_control_page.dart

```diff
@@ -0,0 +1,199 @@
+import 'dart:async';
+
+import 'package:after_layout/after_layout.dart';
+import 'package:autonomy_flutter/common/injector.dart';
+import 'package:autonomy_flutter/main.dart';
+import 'package:autonomy_flutter/screen/app_router.dart';
+import 'package:autonomy_flutter/screen/detail/preview/touchpad_page.dart';
+import 'package:autonomy_flutter/service/canvas_client_service.dart';
+import 'package:autonomy_flutter/util/style.dart';
+import 'package:autonomy_flutter/view/artwork_common_widget.dart';
+import 'package:autonomy_flutter/view/responsive.dart';
+import 'package:autonomy_flutter/view/touchpad.dart';
+import 'package:autonomy_theme/autonomy_theme.dart';
+import 'package:autonomy_tv_proto/autonomy_tv_proto.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_keyboard_visibility/flutter_keyboard_visibility.dart';
+import 'package:flutter_widget_from_html/flutter_widget_from_html.dart';
+import 'package:nft_collection/models/asset_token.dart';
+
+class KeyboardControlPagePayload {
+  final AssetToken assetToken;
+  final CanvasDevice device;
+
+  KeyboardControlPagePayload(this.assetToken, this.device);
+}
+
+class KeyboardControlPage extends StatefulWidget {
+  final KeyboardControlPagePayload payload;
+
+  const KeyboardControlPage({super.key, required this.payload});
+
+  @override
+  State<StatefulWidget> createState() => _KeyboardControlPageState();
+}
+
+class _KeyboardControlPageState extends State<KeyboardControlPage>
+    with AfterLayoutMixin, WidgetsBindingObserver, RouteAware {
+  final _focusNode = FocusNode();
+  late Timer? _timer;
+  final _controller = KeyboardVisibilityController();
+  late StreamSubscription? _keyboardSubscription;
+  final _textController = TextEditingController();
+  final _scrollController = ScrollController();
+  bool _isExpanded = false;
+
+  @override
+  void initState() {
+    super.initState();
+    WidgetsBinding.instance.addObserver(this);
+  }
+
+  @override
+  void dispose() {
+    _timer?.cancel();
+    _textController.dispose();
+    _keyboardSubscription?.cancel();
+    WidgetsBinding.instance.removeObserver(this);
+    routeObserver.unsubscribe(this);
+
+    super.dispose();
+  }
+
+  @override
+  void didPopNext() {
+    super.didPopNext();
+    setState(() {
+      _isExpanded = false;
+    });
+  }
+
+  @override
+  void didPushNext() {
+    super.didPushNext();
+    setState(() {
+      _isExpanded = true;
+    });
+  }
+
+  @override
+  void afterFirstLayout(BuildContext context) {
+    showKeyboard();
+    _keyboardSubscription = _controller.onChange.listen((bool isVisible) {
+      if (!isVisible && !_isExpanded) {
+        Navigator.of(context).pop();
+      }
+    });
+    bool scrolledToTop = false;
+    _timer = Timer.periodic(const Duration(milliseconds: 500), (timer) {
```

Comment:
look pretty complicated, can you describe the scenario about this case a litte bit?